### PR TITLE
Add portage support for Gentoo Linux

### PR DESCRIPTION
**Pull request by zmedico**
Closes: https://github.com/anchore/syft/issues/998

Signed-off-by: Zac Medico <zachary.medico@sony.com>

### CODE REVIEW

**Comment by spiffcs** [comment]
👋 Thanks @zmedico for the PR here. I'm looking at this now. Do you mind if I add some commits to the branch to get the different CI checks passing?

I really appreciate you updating the branch recently. Apologies for merging in other PR which might have broken yours.

**Comment by zmedico** [comment]
> wave Thanks @zmedico for the PR here. I'm looking at this now. Do you mind if I add some commits to the branch to get the different CI checks passing?

Yes, please do.

> I really appreciate you updating the branch recently. Apologies for merging in other PR which might have broken yours.

No problem.

**Review by spiffcs — APPROVED** [review]
This looks good and approved! Just looking if you have any documentation or cross-reference for the regular expression you included for the `cpvRe`

Any path or code reference to the Gentoo Portage db specification will help here for future developers.

**Comment by spiffcs** [comment]
I'll update the CLI tests here too

File: syft/pkg/cataloger/portage/cataloger.go [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+	"github.com/anchore/syft/internal"
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/file"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/source"
+)
+
+var (
+	cpvRe = regexp.MustCompile(`/([^/]*/[\w+][\w+-]*)-((\d+)((\.\d+)*)([a-z]?)((_(pre|p|beta|alpha|rc)\d*)*)(-r\d+)?)/CONTENTS$`)
```
[the remark this answers is not in the document]
Comment (zmedico) [reply syft/pkg/cataloger/portage/cataloger.go:25]: It's derived from the `_cpv` regular expression found in this file:

https://github.com/gentoo/portage/blob/master/lib/portage/versions.py

**Pull request merged**